src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/VersionsController.kt

```diff
@@ -0,0 +1,36 @@
+package org.imperial.mrc.hint.controllers
+
+import org.imperial.mrc.hint.db.SnapshotRepository
+import org.imperial.mrc.hint.db.VersionRepository
+import org.imperial.mrc.hint.exceptions.VersionException
+import org.imperial.mrc.hint.models.SuccessResponse
+import org.imperial.mrc.hint.models.Version
+import org.imperial.mrc.hint.models.asResponseEntity
+import org.imperial.mrc.hint.security.Session
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.bind.annotation.PostMapping
+
+@RestController
+class VersionsController(private val session: Session,
+                         private val snapshotRepository: SnapshotRepository,
+                         private val versionRepository: VersionRepository)
+{
+    @PostMapping("/version/")
+    @ResponseBody
+    fun newVersion(@RequestBody request: Map<String, String>): ResponseEntity<String>
+    {
+        val versionName = request["name"] ?: throw VersionException("Version name missing")
```

Comment:
Oops, yes, I forgot about that. I guess for errors which shouldn't come about in the normal operation of the application we can probably get away without translations - the front end shouldn't be sending missing or empty names here. And as you say they're probably not so helpful anyway!

Reply:
I think this is better as a path variable as we''re going to end up with a few /version/{name}/snapshot/{id}.. endpoints so this will be more consistent.

Reply:
It's rather unorthodox to have a path variable that doesn't exist yet. `/version/name/whatever` makes sense when `name` exists already, but for a POST to create, I think the request body param is the correct approach!

Reply:
OK, well then I don't think RequestParam works for us, isn't that just for query string params? Could make a little data class that's expected to be found  in the @RequestBody and just contains a name property

Reply:
I don't think so, check out the integration tests for https://github.com/mrc-ide/hint/blob/master/src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/PasswordController.kt#L56 - the params are passed in the POST request body, not as query strings.

Reply:
Ah ok, that's form data rather than json, which wasn't working for me. I'll use form data too.

Reply:
https://github.com/mrc-ide/hint/blob/master/src/app/src/test/kotlin/org/imperial/mrc/hint/integration/PasswordTests.kt#L64